clang/lib/CIR/CodeGen/Address.h

```diff
@@ -52,6 +52,14 @@ class Address {
            elementType);
   }
 
+  Address(mlir::Value pointer, clang::CharUnits alignment)
+      : Address(pointer,
+                mlir::cast<cir::PointerType>(pointer.getType()).getPointee(),
```

Comment:
I see we're doing this cast to `cir::PointerType` a few places here without any sort of checking to ensure that this is a `cir::PointerType`.  Is there any reason that this SHOULDN'T take/store a `cir::PointerType` instead of a `mlir::Value`?

Reply:
The thing that's being cast is `pointer.getType()` and `pointer` is an `mlir::Value` which can basically be any operation that has a pointer type. There is no common base class other than `mlir::Value` that can be used here.

Reply:
Got it, thanks for the clarification!  NVM then.